src/components/Navbar/Navbar.js

```diff
@@ -1,42 +1,63 @@
 //Dependencies
-import React from 'react';
+import React, { useState } from 'react';
 import { Link } from 'react-router-dom';
+import { useSelector } from 'react-redux';
 
+//Components, assets, actions, styles etc..
+import NavItem from './NavItem';
+import { ReactComponent as HUOALogo } from '../../assets/huoa-logo.svg';
 import './Navbar.scss';
 
-class Navbar extends React.Component {
-  render() {
-    return (
-      <div>
+const Navbar = () => {
+  const [ navState, setNavState ] = useState(false);
+  const { routes } = useSelector((state) => state.router);
+
+  const renderMenuItems = (children) => {
+    return children.map((item) => (
+      <Link
+        className="navbar__dropdown--link"
+        to={item.linkTo}
+        onClick={() => setNavState(false)}
+        key={item.linkTo}
+        data-text={item.name}
+      >
+        {item.name}
+      </Link>
+    ));
+  };
+
+  const renderNavItems = (items) => {
+    return items.map((item) => {
+      return (
+        <NavItem name={item.name} linkTo={item.linkTo} key={item.linkTo} navState={navState} setNavState={setNavState}>
+          {!!item.children && (
+            <div className="navbar__dropdown">
+              <ul className="navbar__dropdown--links">{renderMenuItems(item.children)}</ul>
+            </div>
+          )}
+        </NavItem>
+      );
+    });
+  };
```

Comment:
While it's working, it's not good practice to nest <a> within another <a>. See below for reference:

[https://stackoverflow.com/questions/9882916/are-you-allowed-to-nest-a-link-inside-of-a-link](url)

There's a few work arounds you can do. Let me know if you'd like to talk it through.

Reply:
Good catch mike

Reply:
Presto, I found a fix, lmk what you think